plugin/pkg/admission/priority/admission.go

```diff
@@ -212,7 +212,10 @@ func (p *priorityPlugin) admitPod(a admission.Attributes) error {
 			}
 
 			priority = pc.Value
-			preempting = pc.Preempting
+
+			if utilfeature.DefaultFeatureGate.Enabled(features.NonPreemptingPriority) && pc.Preempting != nil && pc.Preempting != preempting {
```

Comment:
It should be enough to check the feature gate only. Pods are not expected to set `Preempting` field directly.